slatedb/src/batch_write.rs

```diff
@@ -121,12 +121,34 @@ impl DbInner {
     ) -> Option<tokio::task::JoinHandle<Result<(), SlateDBError>>> {
         let this = Arc::clone(self);
         let mut is_stopped = false;
+        let mut is_first_write = true;
+
+        /// Monitors the the first write's durability watcher. If it takes longer than
+        /// 5 seconds to become durable, log a warning.
+        async fn monitor_first_write(
```

Comment:
nit: how about have this function in a lambda like `let monitor_first_write = async |durability_watcher| { ..`? in L142 we had an lambda `let fut = async move {`, maybe it'd not be bad to have them in consistent.